### PR TITLE
OADP-660: Add azure support

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
related Operator PR: https://github.com/openshift/oadp-operator/pull/780

### CODE REVIEW

File: controllers/common.go [hunk 1]
```diff
@@ -80,36 +89,66 @@ func PopulateResticSecret(name string, namespace string, label string) (*corev1.
...
+		resticSecretData := &corev1.Secret{
+			Data: map[string][]byte{
+				AzureAccountName: AzureAccountNameValue,
+				AzureAccountKey:  AzureAccountKeyValue,
+				ResticPassword:   ResticPasswordValue,
+				ResticRepository: []byte(resticrepo),
+			},
+		}
+		secret.Data = resticSecretData.Data
+		return nil
```
Comment (kaovilai) [inline controllers/common.go:146]: - Add a case for empty provider labe
- Add a default case for unsupported providers

File: controllers/common.go [hunk 2]
```diff
@@ -118,30 +157,63 @@ func ValidateResticSecret(resticsecret *corev1.Secret) error {
...
+				}
+			case ResticRepository:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("resticRepository value cannot be empty")
+				}
 			}
 		}
 	}
+
```
Comment (kaovilai) [inline controllers/common.go:216]: - Add a case for empty provider labe
- Add a default case for unsupported providers

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: controllers/common.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/common.go:146]: @kaovilai we do not support empty provider or any other provider other than aws/azure/gcp, its already validated in bsl.go https://github.com/openshift/oadp-operator/blob/master/controllers/bsl.go#L61

File: controllers/common.go (hunk 2 above)
Comment (shubham-pampattiwar) [reply controllers/common.go:216]: @kaovilai we do not support empty provider or any other provider other than aws/azure/gcp, its already validated in bsl.go https://github.com/openshift/oadp-operator/blob/master/controllers/bsl.go#L61

**Review by savitharaghunathan — COMMENTED** [review]
Visual ack
/lgtm

File: controllers/common.go (hunk 2 above)
Comment (kaovilai) [reply controllers/common.go:216]: Are we absolutely positive that no one else will be playing around with labels?

I see that we like to have error handling done in one place. Sometimes that's refactored away and then you'll hit issues.

Comment (kaovilai) [reply controllers/common.go:216]: We've merged with assumptions of a prior validation in another function before so I'm not opposed to continuing the trend.. but it just itches me a little bit.

**Review by kaovilai — APPROVED** [review]

**Review by savitharaghunathan — APPROVED** [review]
Tested back/restore workflow and it worked.

**Review by eemcmullan — APPROVED** [review]

**Pull request merged**